68 landing pages filter and search

emullaney opened the pull request:
Search and filter functionality on Case Study Index page only for now. I'd like @kmtracey to check this over and make comments before I work on Marketplace Index page.

Not sure on the UI quite yet, might need some more enhancements to make it more user friendly. Also needs an expand/collapse image, I'm using another image as a placeholder for now.

Commit:
refs #68 wip: case study index page filters

Commit:
refs #68 filters with collapsible headers

Commit:
refs #68 adding search functionality

Commit:
#68 added listing icon

Commit:
refs #68 wip inclusion tag for filter anchors

Commit:
refs #68 merging in css conflicts

Commit:
refs #68 inclusion tags!!

Commit:
refs #68 only display tag filters if tags exist, ie. if no regions exist on any case studies, do not show regions

Commit:
refs #68 some minor styling around pagination

Commit:
refs #68 removing comma stripping from CaseStudyIndexPage model get_context method

Commit:
refs #68 send through search query post after new filters are clicked, working on marketplace landing

Commit:
refs #68 adding case study landing page with filters and search

kmtracey commented on portal_pages/templatetags/portal_extras.py:
```diff
@@ -11,6 +11,66 @@ def make_unique(value, unique_var): # Only one argument.
+    
+    return {
+        'request': context['request'],
+    }
```
There is no need to use an existing context object and annotate it to pass back in to render the inclusion tag's template. In this case it would make more sense to return a dict from this function that has "tags_header", "collapse_state", and "link_items" rather than annotating request with these things. Simplifies both the code and the template...

kmtracey commented on portal_pages/templatetags/portal_extras.py:
```diff
@@ -11,6 +11,66 @@ def make_unique(value, unique_var): # Only one argument.
+
+    context.request.tags_header = tags_header # ie. countries
+    context.request.tags_header_title = tags_header.title() # ie. Countries
+    
```
I'm not sure what issue you were seeing with accessing stuff in the context, the normal pattern for accessing request in context here would be to do something like "request = context['request']" and then proceed to use the request local variable in the function, which would remove a lot of repetitive context. prefixes on request below...

kmtracey commented on portal_pages/templates/portal_pages/marketplace_index_page.html:
```diff
@@ -1,67 +1,116 @@
+
+{% endblock %}
 
+{% block extra_js %}
```
We should avoid duplicating this block in multiple templates...

kmtracey committed:
Merge branch 'develop' into 68-landing-pages-filter-and-search

(Resolved) Conflicts:
	rapidpro_community_portal/templates/base.html

kmtracey committed:
Refs #68: filter/search improvements

Factor out some common code/templates instead of duplicating,
minor linter fixes, make user-visible references to Organization
use British English spelling.

kmtracey committed:
Refs #68 minor changes from code review with Erin

emullaney commented:
:+1:
